Entities/AtomGroup.cpp

```diff
@@ -2035,17 +1998,18 @@ before adding them to the MovableMan.
 //                  halted = m_pOwnerMO->OnBounce(position);
 
                 // Calculate the distributed mass that each bouncing Atom has.
-                massDist = mass / (hitTerrAtoms.size()/* + atomsHitMOsCount*/) *
-                                (m_Resolution ? m_Resolution : 1);
+                massDist = mass / static_cast<float>((hitTerrAtoms.size()/* + atomsHitMOsCount*/) *
+                                (m_Resolution ? m_Resolution : 1));
 
                 // Gather the collision response effects so that the impulse force can be calculated.
-                for (aoItr = hitTerrAtoms.begin(); aoItr != hitTerrAtoms.end(); ++aoItr)
+                for (pair<Atom *, Vector> aoItr : hitTerrAtoms)
```

Comment:
Same variable naming issue here. Also, you should pass this by reference (probably const), otherwise I'm pretty sure it's making a copy of the pair each time.
In terms of names, I usually go with `containernameEntry` for these cases, so `const pair<Atom *, Vector> &hitTerrAtomsEntry`. If you've got a better name, feel free to use it though.

Reply:
Yeah, good points.